### DISCUSSION TITLE
FNIRSI GC-01: Restore

### DISCUSSION THREAD

**Discussion by ChyrkaA** [Gissio/radpro#133] [Q&A] [answered]
Please help me restore my FNIRSI GC-01 device

Immediately after the purchase, I wanted to update my device to version 1.6. Downloaded the official firmware from the fnirsi website [https://www.fnirsi.com/cdn/shop/files/GC01_V1.6.bin?v=15824974501055980857](https://www.fnirsi.com/cdn/shop/files/GC01_V1.6.bin?v=15824974501055980857)

I downloaded this file, connected it to a computer with Windows 11 via USB, the computer sees the connected device in the device manager as a USB Mass storage device, but the disk did not appear in Explorer. The same story happened with Windows 10. I inserted it into different ports and different cables - the result was the same.

Then I connected this device to the macBook and miraculously, the disk was mounted. The disk seemed to be 66 kilobytes in size, but the firmware was 64 kilobytes in size. But all the same, the firmware did not want to be copied to the mounted disk of the device and an error occurred that there was not enough space.

The third time, I managed to copy the firmware, ignoring messages about insufficient space. The firmware file has appeared on the device. I disconnected the device and that's it. It didn't turn on any further. I tried to download firmware version 1.6 again, but there was an error about insufficient space.

I tried to upload firmware versions 0.3, 1.5, and even RadPRO - the device still does not turn on. Then I connected the device to a computer with Ubuntu 22.04, the device is displayed, and there is now 85 kilobytes of free space in it. I tried downloading firmware version 1.6 again without any difficulty. But to no avail.

I made different combinations, uploaded the firmware, then pressed Reset, turned on the device with the side button pressed. Nothing helped. The device still doesn’t turn on(((I have a processor on the ch32f10 device. Out of despair, I even tried to upload the firmware from apm. But nothing. I tried to upload the bootloader. Also nothing. The only thing is, after updating the firmware, sometimes the following files appear on the device: errapp.txt, unknown.txt and readme.txt (precisely readme.txt, not ready.txt, like many people on the forums).

I read here that you can upload the firmware once, and then again and overwrite it on the device. But Ubuntu does not allow me to do this, it says that there is not enough space on the device. When I ignore this error, the firmware file, for example, writes only 32 of the required 48 kilobytes.

The charging or connection indicator at the bottom glows red and sometimes green.

Friends, help, I beg you. What should I do? It's a shame about the device(((

**Comment by gc01noob** [comment]
1. don't press any button when it starts up, just connect it to a Linux like Ubuntu - what's the drive called? mine is "BOOTLOADER". the .txt files are system notifications eg apperr.txt means it failed - keep resetting till is says something like ready.txt
2. you'll find in the repo document folder a 1.6.1 version of the original firmware - this sorted mine when i bricked it many times at first when using Windows - make sure you do a full reset before every attempt and disconnect/reconnect - do not unmount/powercycle, fully remove the cable every time
3. make sure you are using the correct MCU version of the firmware
4. don't be afraid to reset the crap out of it until it complies
5. cables are critical, try every one you own.

seems there might be some non-complaint/maybe-fake versions out there - this is the 2nd this week

I pulled my hair out for days trying different methods and resetting with about 3 different .txt errors before i found that the copy twice/overwrite under Ubuntu GUI worked.
takign notes of everything you try as you try it helps a lot as it's easy to get confused.

if all else fails the STM flasher should be your best option - they're on ebay for around 10$/£

to help Gissio can you post the full specs of your device ie remove the case and post the actual MCU, screen and tube codes

**Reply by ChyrkaA** [reply]
Thanks for the answer. Below is a photo of the insides of my device. In the second photo the tube is unfolded; the inscription is not visible, but the marking is J321
![photo_2024-08-20_21-58-13 (2)](https://github.com/user-attachments/assets/04ff31e5-3a76-4cdc-8359-69e48f9dfe30)
![photo_2024-08-20_21-58-13](https://github.com/user-attachments/assets/a095be22-a360-4feb-b4ac-ed0a2f25a109)

Now I have connected the device again. Here is the file on it:
![Знімок екрана з 2024-08-20 21-51-28](https://github.com/user-attachments/assets/3b35900e-1328-4fd8-9d59-aa60f8c73ed2)


In the list of files for my chip, I found only version 1.5 firmware. 

![Знімок екрана з 2024-08-20 21-49-12](https://github.com/user-attachments/assets/5bd04b63-c6b0-4642-ba0d-217aaaee569c)

I downloaded it and uploaded it to my device. Then I disconnected the device from the cable - the device still did not turn on, I did not touch the reset button. Then I connected the device again and there is this file:


![Знімок екрана з 2024-08-20 21-54-10](https://github.com/user-attachments/assets/744b6486-ab0a-45a7-a366-541adaa03ab1)

**Reply by ChyrkaA** [reply]
> 1. don't press any button when it starts up, just connect it to a Linux like Ubuntu - what's the drive called? mine is "BOOTLOADER". the .txt files are system notifications eg apperr.txt means it failed - keep resetting till is says something like ready.txt
> 2. you'll find in the repo document folder a 1.6.1 version of the original firmware - this sorted mine when i bricked it many times at first when using Windows - make sure you do a full reset before every attempt and disconnect/reconnect - do not unmount/powercycle, fully remove the cable every time
> 3. make sure you are using the correct MCU version of the firmware
> 4. don't be afraid to reset the crap out of it until it complies
> 5. cables are critical, try every one you own.
> 
> seems there might be some non-complaint/maybe-fake versions out there - this is the 2nd this week
> 
> I pulled my hair out for days trying different methods and resetting with about 3 different .txt errors before i found that the copy twice/overwrite under Ubuntu GUI worked. takign notes of everything you try as you try it helps a lot as it's easy to get confused.
> 
> if all else fails the STM flasher should be your best option - they're on ebay for around 10$/£
> 
> to help Gissio can you post the full specs of your device ie remove the case and post the actual MCU, screen and tube codes

Hi. How can I contact Gissio?

**Comment by gc01noob** [comment]
btw might be worth checking any hidden files on the drive and checking/editing any filenames to ensure they all match/overwrite.
don't touch the .txt it's a system message file and changes to report system status

**Reply by gc01noob** [reply]
ps. might be readme.txt and not ready.txt to mean it's ok - it's been a while....the other 2 are definitely errors so do a reset if it shows them unti they clear

always disconnect, reset then reconnect before trying anything new and never power it on while pressing any buttons or it will always fail. disk size doesn't report anything meaningful.
also doesn't hurt to reflash the repo 1.6 version between attempts

sorry can't help more, at the end all i needed to do was:
after resetting:
1. connect to Ubuntu with the best cable i had
2. power on
3. drag the .bin file, leave the window open
4. drag the .bin again to the same window
5. ok to overwrite
6. radpro then boots up

**Reply by ChyrkaA** [reply]
After installing firmware 1.5, when I connect the device to the computer, I need to press the Power button for the device to appear in Ubuntu.

How do you manage to overwrite a file if the remaining free space does not allow you to do so?

**Reply by gc01noob** [reply]
it doesn't report a correct drive size.
power it on before connecting.

**Reply by ChyrkaA** [reply]
How can I turn on the device if it does not show any signs of life without connecting to a computer? Or did I understand something wrong?

**Comment by gc01noob** [comment]
i see you get the "GC01 BOOT" drive....hmmmm same as the chap the other day
mine was the apm MCU and it gives "BOOTLOADER" and dragging twice worked.....

sorry can't help any more than to say keep trying things, maybe Gissio can shed some light on the matter, can take a while till he checks the page but he's knows the devices well.

**Reply by ChyrkaA** [reply]
Thanks for your time

**Comment by gc01noob** [comment]
if it says unknown.txt you need to disconnect and reset till it says readme.txt

**Reply by ChyrkaA** [reply]
Unknown appeared after firmware version 1.5

**Reply by gc01noob** [reply]
reset and try another one eg 0.3 till it says readme.txt

**Reply by ChyrkaA** [reply]
0.3 Unknow too(( 
Just checked

**Comment by mressl-itba** [comment]
The firmware on FNIRSI's website is for GC-01's with an APM32 microprocessor.

You need to install the firmware for GC-01's with a CH32 microprocessor, which you can get here:

https://github.com/Gissio/radpro/tree/main/docs/devices/FNIRSI%20GC-01/firmware

Unfortunately there isn't anything else I can do to help you.

**Reply by ChyrkaA** [reply]
> The firmware on FNIRSI's website is for GC-01's with an APM32 microprocessor.
> 
> You need to install the firmware for GC-01's with a CH32 microprocessor, which you can get here:
> 
> https://github.com/Gissio/radpro/tree/main/docs/devices/FNIRSI%20GC-01/firmware
> 
> Unfortunately there isn't anything else I can do to help you.

Thanks for the answer. But I wrote above that I uploaded the firmware specifically for my chip

**Comment by alfmck** [comment]
As I can see from your photo, CH32 from WCH company is used, so you have to use v1.5 file and NOT 1.6 file which is for APM type of processors.
Now, when you killed the soft in your device, try to connect it to win11 or win10 PC and it should be seen on explorer. Then load the 1.5 soft to device.

**Reply by ChyrkaA** [reply]
> As I can see from your photo, CH32 from WCH company is used, so you have to use v1.5 file and NOT 1.6 file which is for APM type of processors.
> Now, when you killed the soft in your device, try to connect it to win11 or win10 PC and it should be seen on explorer. Then load the 1.5 soft to device.

Hello. Thanks for the answer. As I already wrote above, the device is detected in Windows as a USB mass storage device, but is not displayed in Explorer. I tried connecting different cables, different computers, different ports. The result is the same. The device is detected only on Ubuntu and macOs. But even in this case, I already tried to download firmware version 1.5 to the device - but the device still does not turn on(

It's a shame that the Fnirsi website does not indicate which architecture the firmware is for.

**Reply by alfmck** [reply]
You are right, some of GC-01 devices not seen in explorer, while you don't kill the soft in it. As I understand, your device doesn't work now, so the soft is killed. NOW you make reset, connect the device to PC with win10 or win11 again and switch on the device. Then you should see your device in explorer.

**Reply by ChyrkaA** [reply]
> You are right, some of GC-01 devices not seen in explorer, while you don't kill the soft in it. As I understand, your device doesn't work now, so the soft is killed. NOW you make reset, connect the device to PC with win10 or win11 again and switch on the device. Then you should see your device in explorer.

Thanks for the answer. This doesn't help either. The device is detected in win10/11, but is still not displayed in Explorer(((

**Reply by alfmck** [reply]
Some people wrote that they could connect thru USB2 only. For my device in order to connect to win11 PC, I have to:
1. Connect device to PC with win7.
2. Then I see it in explorer. I'm trying to upload any of soft suited for my device. Some time I do it twice. I see an error file in explorer.
3. Then I disconnect it from win7 PC, reset the device, connect to pc with win11, turn on and see it on explorer. After that I upload radpro or original soft suited for CH32 cpu of 48 kB size and see a working picture on LCD. After that device disconnects from explorer.

**Reply by ChyrkaA** [reply]
> Some people wrote that they could connect thru USB2 only. For my device in order to connect to win11 PC, I have to:
> 
> 1. Connect device to PC with win7.
> 2. Then I see it in explorer. I'm trying to upload any of soft suited for my device. Some time I do it twice. I see an error file in explorer.
> 3. Then I disconnect it from win7 PC, reset the device, connect to pc with win11, turn on and see it on explorer. After that I upload radpro or original soft suited for CH32 cpu of 48 kB size and see a working picture on LCD. After that device disconnects from explorer.

I haven't tried win7 yet. I'll try it now and write it down

**Reply by ChyrkaA** [reply]
> Some people wrote that they could connect thru USB2 only. For my device in order to connect to win11 PC, I have to:
> 
> 1. Connect device to PC with win7.
> 2. Then I see it in explorer. I'm trying to upload any of soft suited for my device. Some time I do it twice. I see an error file in explorer.
> 3. Then I disconnect it from win7 PC, reset the device, connect to pc with win11, turn on and see it on explorer. After that I upload radpro or original soft suited for CH32 cpu of 48 kB size and see a working picture on LCD. After that device disconnects from explorer.

no, unfortunately, win7 is the same as win10/11

**Reply by alfmck** [reply]
Strange. You made a reset by pressing microswitch close to USB connector? Please test the battery or charge it.
As you can connect to explorer of Linux or Mac and upload some files, it seems that downloader in MCU works properly.

**Reply by ChyrkaA** [reply]
> Strange. You made a reset by pressing microswitch close to USB connector? Please test the battery or charge it. As you can connect to explorer of Linux or Mac and upload some files, it seems that downloader in MCU works properly.

Now I checked it with a multimeter. CR1220 battery - 2.77 volts, bad battery - 3.4 volts. Is this enough?

**Reply by alfmck** [reply]
And charge the main battery too.

**Reply by ChyrkaA** [reply]
> And charge the main battery too.

CR1220 is good? Only main baterry?

**Reply by ChyrkaA** [reply]
> And charge the main battery too.

No, that doesn't work either. I installed a new battery, tried loading different firmware onto the device - all to no avail(

**Comment by gc01noob** [comment]
fyi the battery condition is critical, you have a bad battery, almost all of them arrive with low batteries that need to be replaced.
sorry I assumed you had checked.
no point continuing or attempting any firmware changes until you have a good battery in there.....

**Reply by ChyrkaA** [reply]
> fyi the battery condition is critical, you have a bad battery, almost all of them arrive with low batteries that need to be replaced. sorry I assumed you had checked. no point continuing or attempting any firmware changes until you have a good battery in there.....


What is the battery itself? CR1220?

**Reply by gc01noob** [reply]
honest advice to make your life easier: forget Windows, use Linux.

**Reply by gc01noob** [reply]
battery type should be printed on it but from memory 1220 (that's the size you can measure with calipers) sounds about right, you can get decent ones on ebay for a just few $/£ - make sure the date is a decent date in the future

**Reply by ChyrkaA** [reply]
> battery type should be printed on it but from memory 1220 (that's the size you can measure with calipers) sounds about right, you can get decent ones on ebay for a just few $/£ - make sure the date is a decent date in the future


So I have a CR1220 - am I normal? Or should it be replaced?

**Reply by ChyrkaA** [reply]
> battery type should be printed on it but from memory 1220 (that's the size you can measure with calipers) sounds about right, you can get decent ones on ebay for a just few $/£ - make sure the date is a decent date in the future

No, that doesn't work either. I installed a new battery, tried loading different firmware onto the device - all to no avail(

**Comment by gc01noob** [comment]
2.7v is bad.

**Reply by ChyrkaA** [reply]
> 2.7v is bad.

Okay, I'll order a new one. How much should it be?

**Reply by gc01noob** [reply]
around this: https://www.ebay.co.uk/itm/114847076325
£2.89 for 2

**Reply by gc01noob** [reply]
if i remember right the 12 means 12mm diameter and the 20 means 2.0mm thick

**Reply by ChyrkaA** [reply]
> if i remember right the 12 means 12mm diameter and the 20 means 2.0mm thick


how many volts should a good cr1220 battery have?

**Reply by gc01noob** [reply]
they're rated 3v (measure up to about 3.2v when new)
so i'd replace when below ~2.9v

**Reply by ChyrkaA** [reply]
> they're rated 3v
> so i'd raplce when below ~2.9


Fine. I'll order a new battery tomorrow, put it in the device and report back. Thank you for your time

**Reply by gc01noob** [reply]
these devices are really sensitive to cr1220 battery voltage so a low battery will still keep the clock time etc but cause havoc with firmware upgrades

**Reply by gc01noob** [reply]
don't give up - sometimes it helps to step back and rethink the strategy and redo steps, can get really frustrating. even worst case the STM/USB programmer is only about £8 on ebay/aliexpress.
imo radpro and radpro-graphing is worth the stress :)

**Reply by gc01noob** [reply]
forgot to also add: between attempts make sure you fully disconnect the USB cable, then reset, then reconnect. unmounting etc doesn't reset it properly.

to recap: I eventually found that with Ubuntu 22.04 Gnome you just need to disconnect the cable, reset, power on normally, connect the best quality cable you have and the drive will mount....then copy twice to the same window and it boots.
I tried everything and bricked it many times till radpro 2.01 worked and now it's great.

https://github.com/Gissio/geigerlog-radpro is awesome too.

**Reply by ChyrkaA** [reply]
> forgot to also add: between attempts make sure you fully disconnect the USB cable, then reset, then reconnect. unmounting etc doesn't reset it properly.
> 
> to recap: I eventually found that with Ubuntu 22.04 Gnome you just need to disconnect the cable, reset, power on normally, connect the best quality cable you have and the drive will mount....then copy twice to the same window and it boots. I tried everything and bricked it many times till radpro 2.01 worked and now it's great.
> 
> https://github.com/Gissio/geigerlog-radpro is awesome too.

I don't understand how you could copy the same thing twice if the Ubuntu operating system prevents you from doing so, showing a message that there is not enough space for another, repeated copy.

**Reply by gc01noob** [reply]
that's the kind of crazy crp that happens with a weak battery or incomplete reset......

**Reply by gc01noob** [reply]
worst case: https://www.ebay.co.uk/itm/266731227085
and ~40p for some header pins.

**Reply by ChyrkaA** [reply]
> that's the kind of crazy crp that happens with a weak battery or incomplete reset......

No, that doesn't work either. I installed a new battery, tried loading different firmware onto the device - all to no avail(

**Comment by Gissio** [comment]
ChyrkaA:

It's not right to demand that I answer your question.

Unfortunately there is only one piece of advice I can give you. Get an ST-LINK V2 programmer (clones work quite well) and flash the firmware with it.

Also: there ~is~ should be no relationship whatsoever between flashing and battery voltage.

**Reply by ChyrkaA** [reply]
> ChyrkaA:
> 
> It's not right to demand that I answer your question.
> 
> Unfortunately there is only one piece of advice I can give you. Get an ST-LINK V2 programmer (clones work quite well) and flash the firmware with it.
> 
> Also: there is no relationship whatsoever between flashing and battery voltage.

Hello. Thank you for your answer. I do not require an answer from you. Sorry if it seemed that way to you) everyone says that you are an expert in this device, so I was really waiting for advice from you. Sorry again

**Reply by alfmck** [reply]
When I've written to test the battery, I had in mind to test the main battery of 1100 mAh. If it is recharged and you connect the device to PC and turn it on, you will not get any reaction from explorer.
It's weird, that you that you can't connect to PC with win7. There was a suggestion, that after connecting to PC you should turn on the device with SHORT press of power on button. 
I had a case, that after flashing the original soft in some gc-01, I couldn't power on the device- LCD was dark. I accidentally pressed LEFT button and keeping it pressed, pushed power on button. The device started and later worked OK.
The interesting flash method is suggested by gc01noob, because it works with your device, and you can make flashing. I can try it with my device, if you want it, please reply.
MY gc-01 uses APM32f103R8T6 from Geehy, which has more flash memory so I can flash every version of soft and they work. In your case you can used soft of 48 kB size ONLY.
I've tried to flash soft to gc-01 using ST-LINK- this method does **not work for me** I get "ERROR: time out while waiting for target halted". But this adaptor works successfully with FS5000 and my APM32f103R8T6 testing board. So, the method with ST-LINK adapter is not 100%, but some people use it for GC-01 flashing, and it is worth to try.

**Reply by ChyrkaA** [reply]
> When I've written to test the battery, I had in mind to test the main battery of 1100 mAh. If it is recharged and you connect the device to PC and turn it on, you will not get any reaction from explorer. It's weird, that you that you can't connect to PC with win7. There was a suggestion, that after connecting to PC you should turn on the device with SHORT press of power on button. I had a case, that after flashing the original soft in some gc-01, I couldn't power on the device- LCD was dark. I accidentally pressed LEFT button and keeping it pressed, pushed power on button. The device started and later worked OK. The interesting flash method is suggested by gc01noob, because it works with your device, and you can make flashing. I can try it with my device, if you want it, please reply. MY gc-01 uses APM32f103R8T6 from Geehy, which has more flash memory so I can flash every version of soft and they work. In your case you can used soft of 48 kB size ONLY. I've tried to flash soft to gc-01 using ST-LINK- this method does **not work for me** I get "ERROR: time out while waiting for target halted". But this adaptor works successfully with FS5000 and my APM32f103R8T6 testing board. So, the method with ST-LINK adapter is not 100%, but some people use it for GC-01 flashing, and it is worth to try.

On Ubuntu, I have 84 kilobytes available on the device, so I can download any firmware.
The battery is good, I checked it. In Windows Explorer, the device is not shown in any way, neither with a short press, nor with a long press, nor with a medium one.
Did you press the left button immediately after flashing the firmware? I tested only with the right button pressed.

Apparently, I’ll have to wait for St-link v2, I ordered it on Aliexpress, and try to flash it with it. I hope it works out. Because if not, then the device, as I understand it, can be thrown away

**Reply by gc01noob** [reply]
tip: when doing unsupported things to undocumented devices be patient, try everything - take notes and don't think things won't work a second time.
you have loads of things to try before you "throw it away" eg try every button, for different time lengths - make sure you disconnect the cable between each action - check that when you press reset that you are actually resetting it. just be careful you don't short anything or physically destroy a component with impatience as then it's a full on repair and not just a re-flash.

ps. mine was also the APM version - the CHS version does seem to be a bit more picky but keep your head and you'll get there

**Reply by alfmck** [reply]
sorry can't help more, at the end all i needed to do was:
after resetting:

connect to Ubuntu with the best cable i had
power on
drag the .bin file, leave the window open **-you will see a bin file icon on explorer window**
drag the .bin again to the same window - **PC says, that such file already is and asks you if you want to replace or keep the old file**
ok to overwrite - **up right place on the message window**
radpro **(or v1.5)** then boots up -**and you will see it on device LCD**
@[ChyrkaA](https://github.com/ChyrkaA)
ChyrkaA
[last week](https://github.com/Gissio/radpro/discussions/133#discussioncomment-10399496)
Author
After installing firmware 1.5, when I connect the device to the computer, I need to press the Power button for the device to appear in Ubuntu. **- not understand - after installing, you disconnect gc-01 and use it**

How do you manage to overwrite a file if the remaining free space does not allow you to do so? _ **new flashing writes data from the beginning of memory, so you will not have to get any messages about remaining free space. Because you choose to overwrite a file, the space be enough.  You SHOULD use bin files of 48kb size only.
**

I tested flashing with Ubuntu. Did precisely as written by gc01noob. Everything works. I feel that you make something wrong. 
 I noticed that after first dragging the bin file, and then resetting and connecting to PC with win11 and power on the device I SEE it in explorer.

**Reply by ChyrkaA** [reply]
> sorry can't help more, at the end all i needed to do was: after resetting:
> 
> connect to Ubuntu with the best cable i had power on drag the .bin file, leave the window open **-you will see a bin file icon on explorer window** drag the .bin again to the same window - **PC says, that such file already is and asks you if you want to replace or keep the old file** ok to overwrite - **up right place on the message window** radpro **(or v1.5)** then boots up -**and you will see it on device LCD** @[ChyrkaA](https://github.com/ChyrkaA) ChyrkaA [last week](https://github.com/Gissio/radpro/discussions/133#discussioncomment-10399496) Author After installing firmware 1.5, when I connect the device to the computer, I need to press the Power button for the device to appear in Ubuntu. **- not understand - after installing, you disconnect gc-01 and use it**
> 
> How do you manage to overwrite a file if the remaining free space does not allow you to do so? _ **new flashing writes data from the beginning of memory, so you will not have to get any messages about remaining free space. Because you choose to overwrite a file, the space be enough. You SHOULD use bin files of 48kb size only. **
> 
> I tested flashing with Ubuntu. Did precisely as written by gc01noob. Everything works. I feel that you make something wrong. I noticed that after first dragging the bin file, and then resetting and connecting to PC with win11 and power on the device I SEE it in explorer.

I can record a video, but ubuntu does not allow me to overwrite the file, since there is no space on the flash memory. Perhaps from the video you can tell me what I'm doing wrong.

**Reply by ChyrkaA** [reply]
> sorry can't help more, at the end all i needed to do was: after resetting:
> 
> connect to Ubuntu with the best cable i had power on drag the .bin file, leave the window open **-you will see a bin file icon on explorer window** drag the .bin again to the same window - **PC says, that such file already is and asks you if you want to replace or keep the old file** ok to overwrite - **up right place on the message window** radpro **(or v1.5)** then boots up -**and you will see it on device LCD** @[ChyrkaA](https://github.com/ChyrkaA) ChyrkaA [last week](https://github.com/Gissio/radpro/discussions/133#discussioncomment-10399496) Author After installing firmware 1.5, when I connect the device to the computer, I need to press the Power button for the device to appear in Ubuntu. **- not understand - after installing, you disconnect gc-01 and use it**
> 
> How do you manage to overwrite a file if the remaining free space does not allow you to do so? _ **new flashing writes data from the beginning of memory, so you will not have to get any messages about remaining free space. Because you choose to overwrite a file, the space be enough. You SHOULD use bin files of 48kb size only. **
> 
> I tested flashing with Ubuntu. Did precisely as written by gc01noob. Everything works. I feel that you make something wrong. I noticed that after first dragging the bin file, and then resetting and connecting to PC with win11 and power on the device I SEE it in explorer.

[Запис з екрана від 2024-08-26 19-29-02.webm](https://github.com/user-attachments/assets/0f3bcbf7-20a8-4b18-a2cb-a9c61033f6f5)

As you can see, the second time the firmware file was not written completely, but only 32 kilobytes out of 48

**Reply by ricsim** [reply]
Simply forget about USB updating!
The ONLY reliable way that ALLWAYS works is using ab ST-Link programmer.
There are some GC-01s that are "nice" and where USB works.
I have one, where there is no chance.
ST-Link allways works, use that if you have issues with USB Update!
Keep in mind that for some (eg Bosean FS-5000) you need to press OK while flashing.
For the GC-01s I have seen this was not necessary.

**Reply by ChyrkaA** [reply]
> Simply forget about USB updating! The ONLY reliable way that ALLWAYS works is using ab ST-Link programmer. There are some GC-01s that are "nice" and where USB works. I have one, where there is no chance. ST-Link allways works, use that if you have issues with USB Update! Keep in mind that for some (eg Bosean FS-5000) you need to press OK while flashing. For the GC-01s I have seen this was not necessary.

Could you tell me how to connect ST-link V2 to the device? Here are the pins on the device:
![photo_2024-08-20_21-58-13](https://github.com/user-attachments/assets/9d427ebe-59e0-46de-9bb3-0c01069e1f84)

The ST-link V2 that I bought is here:
![photo_2024-08-26_22-02-37](https://github.com/user-attachments/assets/0c0d619f-c470-4e72-8803-1703d86fb9d4)

And, in general, will you tell me in as much detail as possible what to do? I have an Ubuntu operating system.
Thank you.

**Reply by alfmck** [reply]
My ubuntu 24.04 LTS, kernel version 68.0-31-generic looks different than in your video. 
GC-01 has name BOOTLOADER, yours GC-01 BOOT. Size - 104.8 kB. 
radpro2.0.2 for APM32 file size - 114.7 kB, but I don't get any warnings about 'not enough memory'
After first flash in properties I see 104.8 total, 114.7 used, 104.7 free.
Have no ideas, maybe ST-LINK really will help.

**Reply by ChyrkaA** [reply]
> My ubuntu 24.04 LTS, kernel version 68.0-31-generic looks different than in your video. GC-01 has name BOOTLOADER, yours GC-01 BOOT. Size - 104.8 kB. radpro2.0.2 for APM32 file size - 114.7 kB, but I don't get any warnings about 'not enough memory' After first flash in properties I see 104.8 total, 114.7 used, 104.7 free. Have no ideas, maybe ST-LINK really will help.

Could you make a video of what it looks like for you to download the firmware onto a device with copying the file twice?

**Reply by ricsim** [reply]
It should be obvious.
3.3v to VCC
GND to GND
SWDIO to DIO
SWCLK to CLK

The square hole by the battery is VCC

**Reply by alfmck** [reply]
[hwlxqyj2utp53x91](https://filebin.net/hwlxqyj2utp53x91)

**Reply by ChyrkaA** [reply]
> The square hole by the battery is VCC

I understand that all this needs to be soldered? "The square hole by the battery is VCC" - what did you mean? As for everything else - I understand

**Reply by ChyrkaA** [reply]
> [hwlxqyj2utp53x91](https://filebin.net/hwlxqyj2utp53x91)

wow! 104 megabytes! now I understand how you manage to copy twice. thanks for the video.

**Reply by ricsim** [reply]
> > The square hole by the battery is VCC
> 
> I understand that all this needs to be soldered? "The square hole by the battery is VCC" - what did you mean? As for everything else - I understand

If you want to do it right, yes then you solder on a header.
I did it quick and dirty by sticking in breadboard pins, and holding them down while flashing. (do not short, and hold them steady while it is flashing!)
One of the holes is more square that marks VCC. forget it it should be completely obvious without that remark as well.

**Reply by ChyrkaA** [reply]
> > > The square hole by the battery is VCC
> > 
> > 
> > I understand that all this needs to be soldered? "The square hole by the battery is VCC" - what did you mean? As for everything else - I understand
> 
> If you want to do it right, yes then you solder on a header. I did it quick and dirty by sticking in breadboard pins, and holding them down while flashing. (do not short, and hold them steady while it is flashing!) One of the holes is more square that marks VCC. forget it it should be completely obvious without that remark as well.

thanks for the answer. ok, then i'm waiting for the programmer itself to arrive. as soon as it's delivered, i'll write to you here, if you don't mind, tell me what to do and what to do

**Reply by ricsim** [reply]
> thanks for the answer. ok, then i'm waiting for the programmer itself to arrive. as soon as it's delivered, i'll write to you here, if you don't mind, tell me what to do and what to do

Absolutely _NO_ need to write here when the programmer is delivered!
Try to flash it, IF you encounter issues, that are not explained in the documentation, and that you cannot resolve elsewere, you could think of writing here.

_I_ am definitely not the right guy to help you, as I also just ordered the programmer for my GC-01s and the Bosean.
I never used it before or after.

Just follow this (and remember that your pinout differs, we already discussed this above, as yours has the pins marked)
https://github.com/Gissio/radpro/blob/main/docs/devices/FNIRSI%20GC-01/install-stlink.md

**Reply by ChyrkaA** [reply]
> > thanks for the answer. ok, then i'm waiting for the programmer itself to arrive. as soon as it's delivered, i'll write to you here, if you don't mind, tell me what to do and what to do
> 
> Absolutely _NO_ need to write here when the programmer is delivered! Try to flash it, IF you encounter issues, that are not explained in the documentation, and that you cannot resolve elsewere, you could think of writing here.
> 
> _I_ am definitely not the right guy to help you, as I also just ordered the programmer for my GC-01s and the Bosean. I never used it before or after.
> 
> Just follow this (and remember that your pinout differs, we already discussed this above, as yours has the pins marked) https://github.com/Gissio/radpro/blob/main/docs/devices/FNIRSI%20GC-01/install-stlink.md

![photo_2024-09-04_11-53-10](https://github.com/user-attachments/assets/189d09da-962e-42cb-9981-324e66e88142)

Tell me, is my device successfully flashed?

**Comment by ChyrkaA** [comment]
![photo_2024-09-04_11-53-10](https://github.com/user-attachments/assets/5783c620-3cbb-4930-91bd-abac235edc20)
Tell me, is my device successfully flashed?

**Reply by ricsim** [reply]
If it rebooted by itself, and is now running Radpro, then yes.
I have no clue about STM programming, but it looks way to short. so i guess it did NOT work.
Try again, and make sure that you follow the guide exactly.

**Comment by Gissio** [comment]
[ChyrkaA](https://github.com/ChyrkaA): It's looking great: your programmer is talking to the GC-01 microprocessor. This means your GC-01  is about to get revived!

Please hang on a while, as there seems to be an issue with the FNIRSI GC-01 flashtool. I'll let you know when I release an update.

**Reply by ChyrkaA** [reply]
> [ChyrkaA](https://github.com/ChyrkaA): It's looking great: your programmer is talking to the GC-01 microprocessor. This means your GC-01 is about to get revived!
> 
> Please hang on a while, as there seems to be an issue with the FNIRSI GC-01 flashtool. I'll let you know when I release an update.

Hello. Thank you for your answer. Okay, I'll wait. Yesterday I checked radpro 2.0.2 - it didn’t work, the device never turned on.

That's why I'm waiting for a new flashtool)) Thank you. You gave me hope!)

**Reply by Gissio** [reply, marked as answer]
Thanks for waiting. ~Here goes a test version of the fixed flashtool~ Use the 2.0.2 v2 flashtool:

https://github.com/Gissio/radpro/releases

Let us know how it worked.

**Reply by ChyrkaA** [reply]
> Thanks for waiting. Here goes a test version of the fixed flashtool:
> 
> [radpro-flashtool-2.0.2-v2.zip](https://github.com/user-attachments/files/16899786/radpro-flashtool-2.0.2-v2.zip)
> 
> Let us know how it worked.


Gissio, it works!!! After two weeks, my device finally worked! Thanks a lot! Should I desolder the programmer wires? Or can I leave it in the case? You never know, maybe in the future, it will be necessary to download more firmware. If they don't affect the background in any way, I'd leave it

**Reply by Gissio** [reply]
I’m glad your counter revived.

It would have been best if you used a 4-pin header instead of soldering the wires directly. But as long as you make sure there is no electrical contact whatsoever, this shouldn’t be a problem.

**Reply by ChyrkaA** [reply]
> I’m glad your counter revived.
> 
> It would have been best if you used a 4-pin header instead of soldering the cables directly. But as long as you make sure there is no electrical contact whatsoever, this shouldn’t be a problem.

and there seems to be no electrical contact with the cable that I soldered to the motherboard of the device. so I leave it in the case so as not to solder it again, because I don't have a connector. 

how do I subscribe to your GC-01 firmware updates? 

do you need a detailed history from me, what I did with the device, from the first failed firmware to how you helped me? maybe this will help you in future developments. and, tell me, why did this happen to me at all? why was the device not flashed by default, like everyone else's)? I'm just wondering what went wrong.

Thank you very much, Gissio, once again. there is a nuance) since my device did not work for a long time, I ordered another one)))))))) now I will have two of them))) the second one will be with the m4011 tube

**Reply by Gissio** [reply]
Go here https://github.com/Gissio/radpro and click on "Watch".

**Comment by ChyrkaA** [comment]
People, has anyone tried replacing the battery with an 18630 ​​on the GC-01 device? Is it worth trying, what do you think?

**Reply by Gissio** [reply]
ChyrkaA: it my understanding that you are a native Ukrainian speaker. Could you check the Ukrainian translation in 2.2beta3?

The strings are available here: https://github.com/Gissio/radpro/blob/main/platform.io/src/strings/uk.h

If you are willing to help us with these, please follow the discussion here: https://github.com/Gissio/radpro/discussions/241
